overlord/fdestate/backend/reseal.go

```diff
@@ -114,49 +127,87 @@ func ResealKeyForBootChains(method device.SealingMethod, rootdir string, params
 	return nil
 }
 
-func resealRunObjectKeys(pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
+func resealRunObjectKeys(updateState stateUpdater, pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
 	// get model parameters from bootchains
 	modelParams, err := boot.SealKeyModelParams(pbc, roleToBlName)
 	if err != nil {
 		return fmt.Errorf("cannot prepare for key resealing: %v", err)
 	}
 
+	numModels := len(modelParams)
+	if numModels < 1 {
+		return fmt.Errorf("at least one set of model-specific parameters is required")
+	}
+
+	pcrProfile, err := secbootBuildPCRProtectionProfile(modelParams)
+	if err != nil {
+		return err
+	}
+
+	var models []secboot.ModelForSealing
+	for _, m := range modelParams {
+		models = append(models, m.Model)
+	}
+
 	// list all the key files to reseal
 	keyFiles := []string{device.DataSealedKeyUnder(boot.InitramfsBootEncryptionKeyDir)}
 
 	resealKeyParams := &secboot.ResealKeysParams{
-		ModelParams:          modelParams,
+		PCRProfile:           pcrProfile,
 		KeyFiles:             keyFiles,
 		TPMPolicyAuthKeyFile: authKeyFile,
 	}
 	if err := secbootResealKeys(resealKeyParams); err != nil {
 		return fmt.Errorf("cannot reseal the encryption key: %v", err)
 	}
 
+	if err := updateState("run+recover", "all", []string{"run", "recover"}, models, pcrProfile); err != nil {
+		return err
+	}
+
 	return nil
 }
 
-func resealFallbackObjectKeys(pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
+func resealFallbackObjectKeys(updateState stateUpdater, pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
 	// get model parameters from bootchains
 	modelParams, err := boot.SealKeyModelParams(pbc, roleToBlName)
 	if err != nil {
 		return fmt.Errorf("cannot prepare for fallback key resealing: %v", err)
 	}
 
+	numModels := len(modelParams)
+	if numModels < 1 {
+		return fmt.Errorf("at least one set of model-specific parameters is required")
+	}
+
+	pcrProfile, err := secbootBuildPCRProtectionProfile(modelParams)
+	if err != nil {
+		return err
+	}
+
+	var models []secboot.ModelForSealing
+	for _, m := range modelParams {
+		models = append(models, m.Model)
+	}
```

Comment:
```suggestion
func modelsFromModelParams() []secboot.ModelForSealing {
	var models []secboot.ModelForSealing
	for _, m := range modelParams {
		models = append(models, m.Model)
	}
	return models
}
```

Reply:
Not sure I understand. Github is probably not showing me the right thing. Do you want me to factor the similar code from `resealFallbackObjectKeys` and `resealRunObjectKeys`?

Reply:
yes, not a big thing though, but both functions are close to each other and maybe it would be a bit nicer